tests/test_cache_lock_file_perf.py

```diff
@@ -0,0 +1,80 @@
+import multiprocessing
+import os
+import time
+
+import pytest
+
+from msal_extensions import FilePersistence, CrossPlatLock
+
+
+@pytest.fixture
+def cache_location():
+    path_to_script = os.path.dirname(os.path.abspath(__file__))
+    return os.path.join(path_to_script, "msal.cache")
+
+
+def _validate_result_in_cache(expected_entry_count, cache_location):
+    with open(cache_location) as handle:
+        data = handle.read()
+    prev_process_id = None
+    count = 0
+    for line in data.split(("\n")):
+        if line:
+            count += 1
+            tag, process_id = line.split(" ")
+            if prev_process_id is not None:
+                assert process_id == prev_process_id, "Process overlap found"
+                assert tag == '>', "Process overlap_found"
+                prev_process_id = None
+            else:
+                assert tag == '<', "Opening bracket not found"
+                prev_process_id = process_id
+
+    assert count == expected_entry_count*2, "No of processes don't match"
```

Comment:
Add an `import logging` at the beginning, and then change this line into:
```suggestion
    assert count <= expected_entry_count*2, "File content corrupted"
    if count < expected_entry_count*2:
        logging.warning("Starvation detected")
```

After that, rerun the test 3~5 times to see if it still "fails". If not, then you can tentatively try ramping up the `num_of_processses` from its current value `10` to `20`, rinse and repeat. And then, perhaps 30, 50, 70..., if it all goes well, you can eventually go all the way up to 100, which was the value used in .Net test. (What number was used by Java, again?)

And then we discuss.